cmd/tsidp/tsidp.go

```diff
@@ -123,11 +131,36 @@ func main() {
 	} else {
 		ts := &tsnet.Server{
 			Hostname: *flagHostname,
-			Dir:      *flagDir,
 		}
 		if *flagVerbose {
 			ts.Logf = log.Printf
 		}
+
+		if *flagState != "" {
+			if isKubeStatePath(*flagState) {
+				if err := validateKubePermissions(*flagState); err != nil {
+					log.Fatalf("tsidp: state is set to be stored in a Kubernetes Secret, but kube permissions validation for the Secret failed: %v", err)
+				}
+			}
+			s, err := store.New(ts.Logf, *flagState)
+			if err != nil {
+				log.Fatalf("Failed to create state store: %v", err)
+			}
+			ts.Store = s
+			// tsrecorder sets Dir for temporary files even with a custom store.
+			// If flagDir is not set, use a default for temp files.
+			if *flagDir == "" {
+				// TODO(maisem): use a better default temp dir?
+				// os.MkdirTemp might be an option but requires cleanup.
+				// For now, let tsnet handle its default if Dir is empty.
+			} else {
+				ts.Dir = *flagDir
+			}
+		} else if *flagDir != "" {
```

Comment:
I think this does not need to be nested in `else` - you seem to be setting `ts.Dir` no matter of whether  `*flagState` is set or no